test/integration/targets/win_user_right/tasks/tests.yml

```diff
@@ -0,0 +1,374 @@
+  assert:
+    that:
+    - remove_right|changed
+    - remove_right.removed == ['{{ansible_hostname}}\Administrator']
+    - remove_right.added == []
+    - remove_right_actual.users == []
+
+- name: remove from right again
+  win_user_right:
+    name: '{{test_win_user_right_name}}'
+    users: ['Administrator', 'Guests', '{{ansible_hostname}}\Users', '.\Backup Operators']
+    action: remove
+  register: remove_right_again
+
+- name: assert remove from right
+  assert:
+    that:
+    - not remove_right_again|changed
+    - remove_right_again.removed == []
+    - remove_right_again.added == []
+
+- name: add to empty right check
+  win_user_right:
+    name: '{{test_win_user_right_name}}'
+    users: ['Administrator', 'Administrators']
+    action: add
+  register: add_right_on_empty_check
+  check_mode: yes
+
+- name: get actual add to empty right check
+  test_get_right:
+    name: '{{test_win_user_right_name}}'
+  register: add_right_on_empty_actual_check
+
+- name: assert add to empty right check
+  assert:
+    that:
+    - add_right_on_empty_check|changed
+    - add_right_on_empty_check.removed == []
+    - add_right_on_empty_check.added == ['{{ansible_hostname}}\Administrator', 'BUILTIN\Administrators']
+    - add_right_on_empty_actual_check.users == []
+
+- name: add to empty right
+  win_user_right:
+    name: '{{test_win_user_right_name}}'
+    users: ['Administrator', 'Administrators']
+    action: add
+  register: add_right_on_empty
+
+- name: get actual add to empty right
+  test_get_right:
+    name: '{{test_win_user_right_name}}'
+  register: add_right_on_empty_actual
+
+- name: assert add to empty right
+  assert:
+    that:
+    - add_right_on_empty|changed
+    - add_right_on_empty.removed == []
+    - add_right_on_empty.added == ['{{ansible_hostname}}\Administrator', 'BUILTIN\Administrators']
+    - add_right_on_empty_actual.users == ['Administrator', 'BUILTIN\Administrators']
+
+- name: add to empty right again
+  win_user_right:
+    name: '{{test_win_user_right_name}}'
+    users: ['Administrator', 'Administrators']
+    action: add
+  register: add_right_on_empty_again
+
+- name: assert add to empty right
+  assert:
+    that:
+    - not add_right_on_empty_again|changed
+    - add_right_on_empty_again.removed == []
+    - add_right_on_empty_again.added == []
+
+- name: add to existing right check
+  win_user_right:
+    name: '{{test_win_user_right_name}}'
+    users: ['Administrator', 'Guests', '{{ansible_hostname}}\Users']
+    action: add
+  register: add_right_on_existing_check
+  check_mode: yes
+
+- name: get actual add to existing right check
+  test_get_right:
+    name: '{{test_win_user_right_name}}'
+  register: add_right_on_existing_actual_check
+
+- name: assert add to existing right check
+  assert:
+    that:
+    - add_right_on_existing_check|changed
+    - add_right_on_existing_check.removed == []
+    - add_right_on_existing_check.added == ["BUILTIN\\Users", "BUILTIN\\Guests"]
```

Comment:
Oh right, Jinja2 doesn't care about single or double quotes, which is making our generic advice a bit more difficult than we planned :-( We still have to document the behavior of Jinja2...

cc: @jhawkesworth